src/migration.rs

```diff
@@ -73,7 +73,27 @@ impl Migration {
                         }
                     }
                     s.push_str(")");
-                }
+                },
+                &mut CreateTableIfNotExists(ref mut t, ref mut cb) => {
+                    if t.meta.has_id {
+                        t.add_column("id", types::primary());
+                    }
+
+                    cb(t); // Run the user code
+                    let vec = t.make::<T>(false);
+                    s.push_str(&T::create_table_if_not_exists(&t.meta.name()));
```

Comment:
So, I'm wondering because the `CreateTable` and `CreateTableIfNotExists` blocks really only differ on this _one_ line.

Maybe it'd be more readable to initially match on `&mut CreateTable(ref mut t, ref mut cb) | &mut CreateTableIfNotExists(ref mut t, ref mut cb)` and then re-match to differentiate that one line:

```rust
&mut CreateTable(ref mut t, ref mut cb) | 
&mut CreateTableIfNotExists(ref mut t, ref mut cb) => {
    // ... shared code

    s.push_str(match change {
        CreateTable(_, _) => &T::create_table(&t.meta.name()),
        CreateTableIfNotExists(_, _) => &T::create_table_if_not_exists(&t.meta.name()),
        _ => unreachable!()
    });
```

At least I think that would work and not look too bad, based on some initial testing on the playground: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=0ff3bd145e9408ff2087a21c32467bd0

What do you think?

Reply:
That is a better approach.  I'll make these changes tonight and push them up.